catalog/rocks/catalog.go

```diff
@@ -314,7 +356,9 @@ type RefManager interface {
 	// ListBranches lists branches
 	ListBranches(ctx context.Context, repositoryID RepositoryID, from BranchID) (BranchIterator, error)
 
-	// GetCommit returns the Commit metadata object for the given CommitID
+	// GetCommit returns the Commit metadata object for the given CommitID.
+	// The given CommitID may be arbitrarily truncated and should return a valid commit as long
+	// as results are not ambiguous
```

Comment:
Isn't that the job of a dereferencer, rather than of `GetCommit`?

Reply:
good question. I don't know. If you feel strongly about this, I'm ok with changing that.